src/include/sof/ipc.h

```diff
@@ -202,14 +202,24 @@ int ipc_dma_trace_send_position(void);
 /* get posn offset by pipeline. */
 int ipc_get_posn_offset(struct ipc *ipc, struct pipeline *pipe);
 
+enum sof_pm_state {
+	SOF_PM_STATE_D0 = 0,	/* Transition to D0, also initial state */
+	SOF_PM_STATE_D3,	/* Transition to D3 is requested */
+	/* Transitions across D0ix substates, incl. D0 are allowed */
+	SOF_PM_STATE_D0ix,
+};
+
 /* private data for IPC */
 struct ipc_data {
 	/* DMA */
 	struct dma *dmac;
 	uint8_t *page_table;
-
 	/* PM */
-	int pm_prepare_D3;	/* do we need to prepare for D3 */
+	uint32_t pm_target_state;	/* SOF_PM_STATE_D.. */
+	/* FW logic for autonomous D0ix<->D0 transitions is disabled*/
+	uint32_t prevent_power_gating;
+	/* FW logic for autonomous clock gating */
+	uint32_t prevent_clock_gating;
```

Comment:
Sorry, I don't get the last two fields. If you set D3 but prevent power gating, this is not D3...
How would the kernel set these two values, based on what?

Reply:
I've added diagrams to better illustrate the idea at https://github.com/orgs/thesofproject/teams/sof-developers/discussions/ Could we switch there for a discussion ?